### ISSUE TITLE
Module crashes after wrong fingerprint Android

### ISSUE THREAD

**Issue by 2vm** [naoufal/react-native-touch-id#97]
When I activated the `TouchID.authenticate()` and scanned the wrong finger (which is not exists on my device) I caught an error, es expected.
``` typescript
private handlePrint = (): void => {
    const { onSuccess, onFail, reason } = this.props;
    TouchID.authenticate(reason).then((r: Object) => {
      onSuccess && onSuccess();
    }).catch((err: Object) => {
      onFail && onFail();
    })
  }
```
But the next time I trying to run the same function the `catch` instantly executing , instead of run it from the beginning.

If I touch the fingerprint scanner, with valid finger, application throws an error:
`
The callback authenticate() exists in module FingerprintAuth, but only one callback may be registered to a function in a native module.
`
Invalid finger will be just ignored.

So is there any solution? Maybe there is some event like `TouchID.close` or smth like this?
Tested on the real device, android 7.0 + EMUI shell.

# ----- upd -----
Seems like it works in release mode. When you touch fingerprint scanner after fail, it cancels process and you can run the function again. But its not really clear I guess.

**Comment by zibs** [comment]
I think this should be fixed in the next release, which will be released any day now.

**Comment by ycb0318** [comment]
I also have this problem yet

**Comment by manuelcaru** [comment]
I have the same problem

**Comment by zibs** [comment]
released now in 4.0.1

**Comment by 2vm** [comment]
@zibs what exactly released? What kind of behavior we got now? Because I don't see any differences about this issue.
